### ISSUE TITLE
USWDS - Bug: Focus trap doesn't consider elements after initialization

### ISSUE THREAD

**Issue by aduth** [uswds/uswds#5830]
### Describe the bug

Currently, the focus trap utility (used in modals and other components) only considers focusable elements at the time of initialization, which can cause unexpected behaviors when elements are added dynamically.

### Steps to reproduce the bug

1. Go to https://codepen.io/aduth/pen/gOyWwEa
2. Click "Open default modal"
3. Click "Add Button" a few times
4. Press <kbd>Tab</kbvd>

### Expected Behavior

Focus shifts to one of the newly added buttons

Instead, focus is stuck on "Add button", because it was the last (only) tabbable element within the modal at the time of initialization.

### Related code

Focusables computed at initialization:

https://github.com/uswds/uswds/blob/98566ec5dc12807a2b0bc36699a033fe3a711d44/packages/uswds-core/src/js/utils/focus-trap.js#L11

### System setup

- USWDS version: 3.8.0 (latest)
- Operating system: macOS 13.6.5
- Browser: Chrome 123

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mahoneycm** [comment]
Hey there @aduth, thank you so much for the codepen! 

Confirming that I am unable to tab to the newly created buttons within the modal after they are initially added. I _was_ able to tab to them after closing and reopening the modal again. It looks like I'm able to do so because the modal triggers call `modal.focusTrap.update(safeActive)`. A solution may be to call this whenever the DOM is manipulated or a child button is clicked within the modal.

I'm curious about the real-world implications of this behavior. For modals, we recommend **not** nesting [multi-step processes](https://designsystem.digital.gov/components/modal/) within modal windows.  When would a user likely come across this behavior? We also use focus trap in header and language selector.

**Comment by mejiaj** [comment]
We should also consider the use of using the `inert` attribute with focus trap (probably re-compute).

[MDN - Inert](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inert)

**Comment by dct2023** [comment]
@uswds/feds Does anyone have bandwidth to review this for Fed Final Review?

**Comment by annepetersen** [comment]
As of right now, I _think_ @heymatthenry and @thisisdano are the only feds that can do tests that require a local code editor / storybook testing path. We should change that long-term, but for the short-term we should be aware of that bottleneck and direct these appropriately.

**Comment by annepetersen** [comment]
Correcting @juliaelman's reassignment here given [this fact](https://github.com/uswds/uswds/issues/5830#issuecomment-2667178294)
